File: cpp/2373. 矩阵中的局部最大值.cpp
class Solution {
public:
    vector<vector<int>> largestLocal(vector<vector<int>>& grid) {
        int n=grid.size(),m=grid[0].size();
        vector<vector<int> > ans(n-2,vector<int>(n-2,0));
        for(int i=0;i<n-2;i++){
            for(int j=0;j<m-2;j++){
                for(int k=i;k<i+3;k++){
                    for(int h=j;h<j+3;h++){
                        ans[i][j]=max(ans[i][j],grid[k][h]);
                    }
                }
            }
        }
        return ans;
    }
};

File: 刷题/golang/2373. 矩阵中的局部最大值.go
func largestLocal(grid [][]int) [][]int {
    n:=len(grid)
    ans:=make([][]int,n-2)
    for i:=0;i<n-2;i++{
        row:=make([]int,n-2)
        for j:=0;j<n-2;j++{
            mx:=grid[i][j]
            for k:=i;k<i+3;k++{
                for h:=j;h<j+3;h++{
                    if grid[k][h]>mx{
                        mx=grid[k][h]
                    }
                }
            }
            row[j]=mx
        }
        ans[i]=row
    }
    return ans
}